src/main/ipc-handlers.ts

```diff
@@ -16,21 +18,35 @@ export const initializeIpcListeners = (): void => {
 	});
 
 	ipcMain.on(IpcMessages.OPEN_AMONG_US_GAME, (_, platformKey: GamePlatform) => {
-		const platform = DefaultGamePlatforms[platformKey];
-
+		const desktop_platform = platform();
 		const error = () => dialog.showErrorBox('Error', 'Could not start the game.');
-
-		if (platform.launchType === PlatformRunType.URI) {
-			// Just open the URI if we can to launch the game
-			// TODO: Try to add error checking here
-			shell.openPath(platform.run);
-		} else if (platform.launchType === PlatformRunType.EXE) {
+		if (desktop_platform === 'win32') {
+			const game_platform = DefaultWindowsGamePlatforms[platformKey];
+			if (game_platform.launchType === PlatformRunType.URI) {
+				// Just open the URI if we can to launch the game
+				// TODO: Try to add error checking here
+				shell.openPath(game_platform.run);
+			} else if (game_platform.launchType === PlatformRunType.EXE) {
+				try {
+					const process = spawn(path.join(game_platform.run, game_platform.exeFile!));
+					process.on('error', error);
+				} catch (e) {
+					error();
+				}
+			}
+		} else if (desktop_platform === 'linux') {
+			const game_platform = DefaultLinuxGamePlatforms[platformKey];
 			try {
-				const process = spawn(path.join(platform.run, platform.exeFile!));
+				const exec = game_platform.run.split(" ");
+				const command = exec[0];
+				exec.shift();
+				const process = spawn(command,exec);
```

Comment:
I _think_ that if we simply set the platforms launchType to a URI the old code to launch the game would have simply worked without any need for change (I tried to write it to be platform independent) but I'm unable to confirm this so I might be wrong. Since we have the URI (basically a link) opening the link _should_ function identically to executing steam with the same argument if a handler is setup for it (which I believe steam does in linux most of the time).

Reply:
@7RST1 would you be able to check and confirm whether this is the case? (You'd also need to change ```run: '/usr/bin/steam steam://rungameid/945360'``` to be simply ```run :'steam://rungameid/945360'``` in GamePlatform.ts)

Reply:
I did try this, and it failed with `Check failed: chdir(current_directory) == 0` in the terminal logs. I couldn't figure out what that meant and I haven't really used URIs before, so I just decided to run the command instead. When I enter the URI in my browser and choose "Steam Runtime" to open it, the game launches.